src/Datastore/DatastoreClient.php

```diff
@@ -0,0 +1,796 @@
+     * property is only possible by first retrieving the entire entity in its
+     * existing state.
+     *
+     * Example:
+     * ```
+     * $keys = [
+     *     $datastore->key('Person', 'Bob'),
+     *     $datastore->key('Person', 'John')
+     * ];
+     *
+     * $entities = [
+     *     $datastore->entity($key[0], ['firstName' => 'Bob']),
+     *     $datastore->entity($key[1], ['firstName' => 'John'])
+     * ];
+     *
+     * $datastore->upsertBatch($entities);
+     * ```
+     *
+     * @param Entity[] $entities The entities to be upserted
+     * @param array $options Configuration options
+     * @return array [Response Body](https://cloud.google.com/datastore/reference/rest/v1/projects/commit#response-body)
+     */
+    public function upsertBatch(array $entities, array $options = [])
+    {
+        $this->operation->mutate('upsert', $entities, Entity::class);
+        return $this->operation->commit($options);
+    }
+
+    /**
+     * Delete an entity
+     *
+     * Example:
+     * ```
+     * $key = $datastore->key('Person', 'Bob']);
+     *
+     * $datastore->delete($key);
+     * ```
+     *
+     * @param Key $key The identifier to delete
+     * @param array $options Configuration options
+     * @return string The updated entity version number
+     */
+    public function delete(Key $key, array $options = [])
+    {
+        $res = $this->deleteBatch([$key], $options);
+        return $this->parseSingleMutationResult($res);
+    }
+
+    /**
+     * Delete multiple entities
+     *
+     * Example:
+     * ```
+     * $keys = [
+     *     $datastore->key('Person', 'Bob'),
+     *     $datastore->key('Person', 'John')
+     * ];
+     *
+     * $datastore->deleteBatch($keys);
+     * ```
+     *
+     * @param Key[] $keys The identifiers to delete
+     * @param array $options Configuration options
+     * @return array [Response Body](https://cloud.google.com/datastore/reference/rest/v1/projects/commit#response-body)
+     */
+    public function deleteBatch(array $keys, array $options = [])
+    {
+        $this->operation->mutate('delete', $keys, Key::class);
+        return $this->operation->commit($options);
+    }
+
+    /**
+     * Retrieve an entity from the datastore
+     *
+     * Example:
+     * ```
+     * $key = $datastore->key('person', 'Bob');
+     *
+     * $entity = $datastore->lookup($key);
+     * if (!is_null($entity)) {
+     *     echo $entity['firstName']; // 'Bob'
+     * }
+     * ```
+     *
+     * @codingStandardsIgnoreStart
+     * @param Key $key $key The identifier to use to locate a desired entity
+     * @param array $options {
+     *     Configuration Options
+     *
+     *     @type string $readConsistency See
+     *           [ReadConsistency](https://cloud.google.com/datastore/reference/rest/v1beta3/ReadOptions#ReadConsistency).
+     *     @type string $className The name of the class to return results as.
```

Comment:
I'm still going through everything, so apologies if I missed something - but it looks like this may not actually be implemented here